src/test/spec/unified_runner/mod.rs

```diff
@@ -0,0 +1,304 @@
+                            .await
+                            .unwrap();
+                    }
+                    "targetedFailPoint"
+                    | "assertSessionTransactionState"
+                    | "assertSessionPinned"
+                    | "assertSessionUnpinned" => panic!("Transactions not implemented"),
+                    "assertDifferentLsidOnLastTwoCommands"
+                    | "assertSameLsidOnLastTwoCommands"
+                    | "assertSessionNotDirty" => panic!("Explicit sessions not implemented"),
+                    "assertCollectionExists" => {
+                        let arguments = operation.arguments.unwrap();
+                        let collection_name = arguments.get_str("collectionName").unwrap();
+                        assert!(get_collection_names(&arguments, &client)
+                            .await
+                            .contains(&collection_name.to_string()));
+                    }
+                    "assertCollectionNotExists" => {
+                        let arguments = operation.arguments.unwrap();
+                        let collection_name = arguments.get_str("collectionName").unwrap();
+                        assert!(!get_collection_names(&arguments, &client)
+                            .await
+                            .contains(&collection_name.to_string()));
+                    }
+                    "assertIndexExists" | "assertIndexNotExists" => {
+                        panic!("Index management not implemented")
+                    }
+                    other => panic!("Unknown test runner operation: {}", other),
+                }
+            } else {
+                let operation = EntityOperation::from_operation(operation).unwrap();
+
+                let result = operation.execute(&entities).await;
+
+                if let Some(id) = operation.save_result_as_entity {
+                    let result = result.clone().unwrap();
+                    entities.insert(id, result.into());
+                }
+
+                if let Some(expect_result) = operation.expect_result {
+                    let result = result.unwrap_or_else(|_| panic!("operation should succeed"));
+                    match result {
+                        Some(result) => {
+                            assert!(results_match(Some(&result), &expect_result));
+                        }
+                        None => {
+                            panic!("expected {}, got {:?}", &expect_result, &result);
+                        }
+                    }
+                } else if let Some(expect_error) = operation.expect_error {
+                    let error = result.unwrap_err();
+                    expect_error.verify_result(error);
+                }
+            }
+        }
+
+        if let Some(_expect_events) = test_case.expect_events {
+            // TODO implement the spec for command monitoring
```

Comment:
What's the rationale for not doing this yet?

Reply:
I'm working on it right now; we discussed in the team channel that it would be better to get team feedback on the general syntax/structure of the POCs sooner rather than later, so in the interest of getting that feedback now I moved this into review while implementing this piece.

Reply:
Ah, okay, sounds good